lib/model/node.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * node.js
+ *
+ * Definition for the Node class. A Node is a composable data structure that
+ * contains a collection of Inputs and Outputs, a list of UI styles, and
+ * arbitrary key-value metadata. Nodes can also contain other nodes.
+ */
+
+/**
+ * Constructor function.
+ *
+ * @param id - The Node's unique identifier.
+ * @param inputs - An array containing the Node's Inputs.
+ * @param outputs - An array containing the Node's Outputs.
+ * @param styles - An array of strings which contain the styles that will be
+ *                 applied to the node in the visualization.
+ * @param metadata - A standard JS object containing key-value metadata.
+ */
+var Node = function(id, inputs, outputs, styles, metadata) {
+	this.id = id || 0;
+	this.inputs = inputs || [];
+	this.outputs = outputs || [];
+	this.styles = styles || [];
+	this.metadata = metadata || {};
+	this.x = 0;
+	this.y = 0;
+	this.auto = true;
+};
+
+/**
+ * Getter method for the id property.
+ */
+Node.prototype.getID = function() {
+	return this.id;
+};
+
+/**
+ * Getter method for returning an xy-coordinate pair representing the Node's
+ * position in 2D space.
+ */
+Node.prototype.getPos = function() {
+	return{
+		x: this.x,
+		y: this.y
+	};
+};
+
+/**
+ * Setter method for setting an xy-coordinate pair representing the Node's
+ * position in 2D space.
+ */
+Node.prototype.setPos = function(x,y) {
```

Comment:
The other objects are immutable because of how they work. Once they are parsed in nothiung really has to change in their state. Node and Port have to change because of their variable position. 
I don't see why we would return a new object for simply changing positions, but thats just me.